src/pybamm/expression_tree/symbol.py

```diff
@@ -994,6 +1001,12 @@ def create_copy(
         children = self._children_for_copying(new_children)
         return self.__class__(self.name, children, domains=self.domains)
 
+    @deprecated(
+        deprecated_in="25.1.0",
+        removed_in="26.0.0",
+        current_version=pybamm.__version__,
+        details="Use `create_copy` instead.",
+    )
```

Comment:
As we now have the decorator to issue the warning, could you drop the warning that exists inside the function?